src/lib.rs

```diff
@@ -134,22 +133,20 @@ define_if_single_runtime_enabled! {
 
     #[cfg(not(feature = "sync"))]
     pub use crate::{
-        client::Client,
+        client::{Client, session::ClientSession},
         coll::Collection,
         cursor::{Cursor, session::{SessionCursor, SessionCursorHandle}},
         db::Database,
     };
 
     #[cfg(feature = "sync")]
     pub(crate) use crate::{
-        client::Client,
+        client::{Client, session::ClientSession},
         coll::Collection,
         cursor::{Cursor, session::{SessionCursor, SessionCursorHandle}},
         db::Database,
     };
 
-    pub use client::session::ClientSession;
```

Comment:
The sync API originally used the same `ClientSession` as async because `ClientSession` didn't have any async components; however, the transactions methods are async so I needed to introduce a new sync `ClientSession`. This is a breaking change but luckily this shouldn't be a problem given the 2.0 (I doubt this will affect many users either) -- we should still call it out in the release notes when this goes out, though.